discovery/azure/azure.go

```diff
@@ -502,5 +502,5 @@ func (client *azureClient) getNetworkInterfaceByID(networkInterfaceID string) (n
 		err = autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
 	}
 
-	return result, nil
+	return result, err
```

Comment:
Thanks for spotting it! It works but I'd prefer to keep the usual flow like this:

```go
  result, err = client.nic.GetResponder(resp)
  if err != nil {
    return result, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
  }

  return result, nil
```

Also L496 is useless and can be removed while you are it.

Reply:
Fixed, thanks for the hint!